src/Native/Unix/System.Native/pal_io.cpp

```diff
@@ -360,6 +363,28 @@ extern "C" int32_t SystemNative_ReadDirR(DIR* dir, void* buffer, int32_t bufferS
 
     // 0 returned with non-null result (guaranteed to be set to entry arg) -> success
     assert(result == entry);
+#else
+    errno = 0;
+    result = readdir(dir);
+
+    //  kernel set errno -> failure
```

Comment:
Can you clarify what you mean?  Whatever readdir does, the result is always assigned to result (unless the program crashes).